client/context/GlobalContext.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { createContext, useContext, useReducer } from "react";
+
+interface SnackbarState {
+    open: boolean,
+    message: string,
+    severity: string
+}
+
+interface GlobalState {
+    // Modal control
+    modalOpenStates: Record<string, boolean>;
+    // Snackbar
+    snackbarState: SnackbarState;
+}
+
+/**
+ * Pass a shared entity to react components,
+ * specifically a way to make api requests.
+ */
+const globalContextParams: GlobalState = {
+    // Modal control
+    modalOpenStates: {},
+    // Snackbar
+    snackbarState: {
+        open: false,
+        message: "",
+        severity: "info",
+    },
+};
+
+const reducerMapping: Record<string, string> = {
+    // Snackbar
+    SET_SNACKBAR_STATE: "snackbarState",
+};
+
+/**
+ * Update the shared states of react components.
+ */
+const globalReducer = (state: GlobalState, { type, payload }: { type: string, payload: unknown}) => {
```

Comment:
I wonder if payload should be `SnackbarState | unknown` to match what it was back in EditorContext. Not sure how helpful that typehint really is, but I guess it's a start that we can work on making more specific over time.

Reply:
What do you think of this type construction?

```ts
type Action =
    | { type: 'OPEN_MODAL'; payload: string }
    | { type: 'CLOSE_MODAL'; payload: string }
    | { type: 'SET_SNACKBAR_STATE'; payload: SnackbarState };

/**
 * Reducer to manage global state updates.
 */
const globalReducer = (state: GlobalState, action: Action): GlobalState => {
```

Reply:
If that works and covers all the currently supported scenarios, I think that sounds great!